contracts/smart-account/modules/PasskeyRegistryModule.sol

```diff
@@ -33,17 +33,17 @@ contract PasskeyRegistryModule is
         uint256 _pubKeyY,
         string calldata _keyId
     ) external override returns (address) {
-        if (
-            smartAccountPassKeys[msg.sender].pubKeyX != 0 &&
-            smartAccountPassKeys[msg.sender].pubKeyY != 0
-        ) {
+        PassKeyId memory passKeyId = smartAccountPassKeys[msg.sender];
```

Comment:
could you please share a gas comparison b/w `PassKeyId memory passKeyId` and `PassKeyId storage passKeyId` ?

Reply:
I tried both gas comparison b/w **PassKeyId memory passKeyId** and **PassKeyId storage passKeyId**, 
**PassKeyId memory passKeyId** costs more gas even from the main code in the repo.
**PassKeyId storage passKeyId** costs less from the main code and the memory code.
<img width="1052" alt="Screenshot 2023-10-22" src="https://github.com/bcnmy/scw-contracts/assets/56640677/4f3a7998-cdf3-4184-9157-9bed9760f604">

After trying the three approaches, I analysed that the **PassKeyId storage passKeyId** is more gas optimised, but I incorrectly submitted the **PassKeyId memory passKeyId**

Here I share the difference between the main repo code, and the last submitted code I pushed
the down table gas optimisation is  for the main repo, and the upper is the **PassKeyId storage passKeyId** one

Reply:
That makes sense, given that with a memory declaration you're forcing the object to be copied to memory when it's not really need.

With the storage pointer, you're essentially saving one computation for calculating the slot at which object is stored, compared to the original implementation.

Reply:
That was the only reason of this PR.

Thank you.